rust/arrow/src/record_batch.rs

```diff
@@ -113,9 +113,30 @@ impl From<&StructArray> for RecordBatch {
     }
 }
 
+impl Into<StructArray> for RecordBatch {
+    fn into(self) -> StructArray {
+         self.schema.fields.iter().zip(self.columns.iter())
+             .map(|t| (t.0.clone(), t.1.clone()))
+            .collect::<Vec<(Field, ArrayRef)>>()
+             .into()
+    }
+}
+
 unsafe impl Send for RecordBatch {}
 unsafe impl Sync for RecordBatch {}
 
+
+/// Definition of record batch reader.
+pub trait RecordBatchReader {
```

Comment:
We have a very similar trait in DataFusion, only it inherits `Sync + Send` too so that it can be passed between threads. Maybe we could do that here too? See https://github.com/apache/arrow/blob/master/rust/datafusion/src/execution/physical_plan/mod.rs#L44-L50

Reply:
I don't think it's safe to make this trait as Send+Sync because the reader used some unsafe data(e.g. MutableBuffer).

Reply:
OK, makes sense.